graphql/resolve/resolver.go

```diff
@@ -484,6 +484,10 @@ func (r *RequestResolver) Resolve(ctx context.Context, gqlReq *schema.Request) *
 	// we can just execute it.
 	switch {
 	case op.IsQuery():
+		resp.Header = make(map[string][]string)
+		if len(op.CacheControl()) != 0 {
+			resp.Header.Set(schema.CacheControlHeader, op.CacheControl())
```

Comment:
don't forget to add  ^^^. 
`Vary: Accept-Encoding`.

You always have to add this header when putting Cache-Control as per the http spec